pandas/core/generic.py

```diff
@@ -11524,13 +11524,37 @@ def cum_func(self, axis=None, skipna=True, *args, **kwargs):
         def block_accum_func(blk_values):
             values = blk_values.T if hasattr(blk_values, "T") else blk_values
 
-            result = nanops.na_accum_func(values, accum_func, skipna=skipna)
+            if is_extension_array_dtype(values.dtype):
+                result = values._accumulate(name, skipna, **kwargs)
+            else:
+                result = nanops.na_accum_func(values, accum_func, skipna=skipna)
```

Comment:
Might want to do this if statement down in `na_accum_func`. So we always call `nanops.na_accum_func(values, accum_func, skipna)` and it's what dispatches based on whether `values` is an EA or not.

Reply:
Yes, we could do this but note the following, in `_accumulate` we explicitly use the name (just like in _reduce, to map infer the accum_func we actually want to use).
This will leave us with two options
1. break similarity with `reduce` and just abandon the `name`. This will imply that in the `_accumulate` signature, we replace `name` by `accum_func` and could get rid of the `**kwargs` as we only have the arguments passed to `na_accum_func`
2. change the API of `na_accum_func`, to include the `name` (as a kwarg?) and further kwargs as well?

@TomAugspurger what would be your favorite solution?

Reply:
Mmm thanks for the explanation. Let's keep it similar to `_reduce` I think.

Reply:
@jreback here is an explanation why I chose to avoid na_accum_func